fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -21,14 +21,15 @@ export function UserDetails({ userId, onCancel }: Props) {
           `https://jsonplaceholder.typicode.com/users/${userId}`
         );
         const data = await response.json();
-        if (response.ok) {
-          setUser(data);
+        setIsLoading(false);
+        if (!response.ok) {
```

Comment:
This check needs to be done _before_ calling `await response.json()` because if the response is not ok, there might not be any JSON to retrieve, or it won't be what you are expecting.